corehq/apps/case_importer/views.py

```diff
@@ -257,3 +264,73 @@ def excel_commit(request, domain):
     request.session.pop(EXCEL_SESSION_ID, None)
 
     return HttpResponseRedirect(base.ImportCases.get_url(domain))
+
+
+def _get_bulk_case_upload_args_from_request(request, domain):
+    try:
+        upload_file = request.FILES["file"]
+        case_type = request.POST["case_type"]
+        search_field = request.POST['search_field']
+        create_new_cases = request.POST.get('create_new_cases') == 'on'
+        search_column = request.POST['search_column']
+        return upload_file, case_type, search_field, create_new_cases, search_column
+    except Exception:
+        raise ImporterError(
+                "Invalid post request."
+                "Submit the form with field 'file' and the required params")
+
+
+@csrf_exempt
+@require_POST
+@api_auth
+@require_can_edit_data
+def bulk_case_upload_api(request, domain, **kwargs):
```

Comment:
This is restricted to only those with access to all locations. I assume thats intentional?

Reply:
That's because of can_edit_data, right? 

And yeah, for now this just mimics the permissions of the normal uploader, so that permissions set is fine., appreciate the confirmation